gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -0,0 +1,485 @@
+		// It was a request for a file, send them there directly.
+		url := joinPath(gcsBaseURL, bucket, object)
+		logger.Printf("redirect to %s", url)
+		http.Redirect(w, r, url, http.StatusTemporaryRedirect)
+		return
+	}
+	dir.Render(w, path)
+}
+
+// splitBucketObject breaks a path into the first part (the bucket), and
+// everything else (the object).
+func splitBucketObject(path string) (string, string) {
+	path = strings.Trim(path, "/")
+	parts := strings.SplitN(path, "/", 2)
+	if len(parts) == 0 {
+		return "", ""
+	}
+	if len(parts) == 1 {
+		return parts[0], ""
+	}
+	return parts[0], parts[1]
+}
+
+// joinPath joins a set of path elements, but does not remove duplicate /
+// characters, making it URL-safe.
+func joinPath(paths ...string) string {
+	return strings.Join(paths, "/")
+}
+
+// dirname returns the logical parent directory of the path.  This is different
+// than path.Split() in that we want dirname("foo/bar/") -> "foo/", whereas
+// path.Split() returns "foo/bar/".
+func dirname(path string) string {
+	leading := ""
+	if strings.HasPrefix(path, "/") {
+		leading = "/"
+	}
+	parts := strings.Split(strings.Trim(path, "/"), "/")
+	if len(parts) > 1 {
+		return leading + strings.Join(parts[0:len(parts)-1], "/") + "/"
+	}
+	return leading
+}
+
+// parseXML extracts a gcsDir object from XML.  If this returns a nil gcsDir,
+// the XML indicated that this was not a directory at all.
+func parseXML(body []byte, object string) (*gcsDir, error) {
+	dir := new(gcsDir)
+	if err := xml.Unmarshal(body, &dir); err != nil {
+		return nil, err
+	}
+	// We think this is a dir if the object is "/" (just the bucket) or if we
+	// find any Contents or CommonPrefixes.
+	isDir := object == "/" || len(dir.Contents)+len(dir.CommonPrefixes) > 0
+	selfIndex := -1
+	for i := range dir.Contents {
+		rec := &dir.Contents[i]
+		name := strings.TrimPrefix(rec.Name, object)
+		if name == "" {
+			selfIndex = i
+			continue
+		}
+		rec.Name = name
+		if strings.HasSuffix(name, "/") {
+			rec.isDir = true
+		}
+	}
+
+	for i := range dir.CommonPrefixes {
+		cp := &dir.CommonPrefixes[i]
+		cp.Prefix = strings.TrimPrefix(cp.Prefix, object)
+	}
+
+	if !isDir {
+		return nil, nil
+	}
+
+	if selfIndex >= 0 {
+		// Strip out the record that indicates this object.
+		dir.Contents = append(dir.Contents[:selfIndex], dir.Contents[selfIndex+1:]...)
+	}
+	return dir, nil
+}
+
+// gcsDir represents a bucket in GCS, decoded from XML.
+type gcsDir struct {
+	XMLName        xml.Name `xml:"ListBucketResult"`
+	Name           string   `xml:"Name"`
+	Prefix         string   `xml:"Prefix"`
+	Marker         string   `xml:"Marker"`
+	NextMarker     string   `xml:"NextMarker"`
+	Contents       []Record `xml:"Contents"`
+	CommonPrefixes []Prefix `xml:"CommonPrefixes"`
+}
+
+func htmlPageHeader(out io.Writer, name string) error {
+	const tmpl = ` 
+	    <!doctype html>
+    	<html>
+    	<head>
+    	    <link rel="stylesheet" href="http://yui.yahooapis.com/pure/0.6.0/pure-min.css">
```

Comment:
huh? that link works for me - https://yui-s.yahooapis.com/pure/0.6.0/pure-min.css

Note the hostname changed (adding a -s) in addition to making it https.